### DISCUSSION TITLE
Erros setup .env document

### DISCUSSION THREAD

**Discussion by cassohir** [outline/outline#2435] [Self Hosting]
I've already set up the .env document. But when I try running the application ($ docker run outlinewiki/outline), I got these messages:

The server could not start, please fix the following configuration errors and try again:

  - The SECRET_KEY env variable must be set with the output of $ OpenSSL rand -hex 32
  - The UTILS_SECRET env variable must be set with a secret value, it is recommended to use the output of $ OpenSSL rand -hex 32
  - The URL env variable must be set to the fully qualified, externally accessible URL, e.g https://wiki.mycompany.com
  - The DATABASE_URL env variable must be set to the location of your Postgres server, including username, password, and port
  - The REDIS_URL env variable must be set to the location of your Redis server, including username, password, and port

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Where should the document be created? I think it's something like this. I cloned the repository and just created the .env file in the main directory. Any Ideas?

**Comment by EacAyf** [comment]
I'd suggest using the docker-compose.yml file as it also creates the other necessary containers like postgres and redis. 
Anyhow you can pass an .env file to your container with the --env-file option. 

docker run --env-file .env outlinewiki/outline 

https://docs.docker.com/engine/reference/commandline/run/
